### DISCUSSION TITLE
Getting ready for 1.0.0 - Suggestions welcome

### DISCUSSION THREAD

**Discussion by yoavniran** [rpldy/react-uploady#194] [General]
<p align="center">
<img src="https://user-images.githubusercontent.com/1102278/121705511-5fdf8780-cadd-11eb-9b6e-afc219ebd6f7.png" height="400"/>
</p>

Uploady has been out for just over  a year now and succeeded far more than I had anticipated in terms of the value it brings and the user-base it gained.

During this time, there have been bugs reported and  features requested. Almost all were small or additive, and none required major overhauls or paradigm shifts. This has been very satisfying for me that the set of features and API decisions didn't need to change since its inception.

Seeing this, Id like to make Uploady officially ready for Production (despite already being used in Production by some). 

> If you have been using it in Production and haven't updated this **[discussion](https://github.com/rpldy/react-uploady/discussions/131)**, I'd love it if you could add your use-case & application/company name there.

--- 

The reason for this post is that I'd like to take the opportunity and ask you, Uploady users, to make your voice heard and **help me** decide if there is anything that should be added to this library before its API and functionality-set is determined as the official 1.0 version.

If you have ideas, requirements, or problems, speak now or forever hold you peace. Ok, not forever. You can always open an issue or a discussion. Major changes may have to wait for 2.0, though.

Thanks!
Yoav

**Comment by selinosherpa** [comment]
This is a minor point but… it seems like half the Storybook links don't lead to the described story.
Uploady is a first-class package and it should have first-class docs to match. 😀

Example
https://react-uploady-storybook.netlify.app/?path=/docs/upload-button--disabled-during-upload

**Reply by yoavniran** [reply]
Thanks @selinosherpa I will look into it. The example link you sent does actually take to the right story. but, it defaults to the `readme` tab, rather than the `canvas` tab. I will look into forcing the canvas tab.

**Reply by selinosherpa** [reply]
Ah, I see now! There's separate stories but all of them share the same documentation so when I switch stories I get the exact same doc as apposed to a doc that describes the story selected. I have not seen that approach before. Thank your for clearing it up. 👍🏾

**Comment by dishcandanty** [comment]
This is probably not the place for it. But I feel like the storybooks are useful to seeing examples, but actually getting code snippets its less than helpful. 

Seems like the helpers are convoluted enough that if I'm just trying to lift one of the explicit examples I have to clone the code base and delete everything except the component i want.

I wish there was just some `code` complete snippets like there are on https://mui.com/

**Reply by yoavniran** [reply]
hi @dishcandanty,
This is great feedback!

The internal storybook evolved into the (integration-)testing ground for Uploady and so I can appreciate it has now become too complex and less useful as a documentation tool.

A big piece in the [1.0.0 plan](https://github.com/rpldy/react-uploady/projects/2) is a dedicated docs site that will contain all the necessary info in a much clearer and ordered fashion. 

Until this in place (will take some time) I recommend these resources:

- [Guides](https://github.com/rpldy/react-uploady/tree/master/guides) section
- [Video tutorials series](https://www.youtube.com/playlist?list=PLigk9hpU4jsqZM7Fr3DxThJ--hsFWG8xk) on youtube
- The [tutorials repo](https://github.com/rpldy/uploady-tutorials) (used in the video series)

Please let me know if there's a specific use-case/flow you're interested in, and feel isnt clear, and I will create a proper tutorial.

Any other feedback is also greatly appreciated.

**Reply by dishcandanty** [reply]
+1 to the dedicated docs site! Guides / Videos are fine. But I would definitely prefer a formatted/navigatabledocs site!

**Comment by yoavniran** [comment]
1.0.0 Released!
